cmd/shim/main_test.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+	"testing/fstest"
+
+	"github.com/dagger/dagger/core"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSecretScrubWriter_Write(t *testing.T) {
+	fsys := fstest.MapFS{
+		"mysecret": &fstest.MapFile{
+			Data: []byte("my secret file"),
+		},
+	}
+	env := []string{
+		"MY_SECRET_ID=my secret value",
+	}
+
+	var buf bytes.Buffer
+	w, err := NewSecretScrubWriter(&buf, fsys, env, core.SecretToScrubInfo{
+		Envs:  []string{"MY_SECRET_ID"},
+		Files: []string{"/mysecret"},
+	})
+	require.NoError(t, err)
+
+	_, err = fmt.Fprintf(w, "I love to share my secret value to my close ones. But I keep my secret file to myself.")
+	require.NoError(t, err)
+	want := "I love to share *** to my close ones. But I keep *** to myself."
+	require.Equal(t, want, buf.String())
+}
+
+func TestFilterSecretToScrub(t *testing.T) {
+	secretValue := "my secret value"
+	env := []string{
+		fmt.Sprintf("MY_SECRET_ID=%s", secretValue),
+		"PUBLIC_STUFF=so public",
+	}
+
+	secretToScrub := core.SecretToScrubInfo{
+		Envs: []string{
+			"MY_SECRET_ID",
+		},
+	}
+
+	secrets := loadSecretsToScrubFromEnv(env, secretToScrub.Envs)
+	require.NotContains(t, secrets, "PUBLIC_STUFF")
+	require.Equal(t, "my secret value", secretValue)
```

Comment:
Seems like this assertion runs against a constant value?

Reply:
Wow, good catch, I guess I messed up updating this test. Will fix it.